src/devices/video/bt45x.cpp

```diff
@@ -0,0 +1,442 @@
+// license:BSD-3-Clause
+// copyright-holders:Patrick Mackinlay
+
+/*
+ * An implementation of the Brooktree Bt45x family (except the Bt459) of RAMDAC
+ * devices. The 453, 454 and 455 are simpler, and have no command or control
+ * registers, whereas the others in the family use these to support additional
+ * options such as multiplexing, blinking etc. Otherwise the devices are quite
+ * similar, and vary only in terms of maximum clock speed, number of color or
+ * overlay bits, and quantity and number bits of resolution of DAC output.
+ *
+ *           Max.       Input          Output
+ *   Part    Clock  Color  Overlay  Num.  Levels  Other
+ *   Bt451  165MHz  8 bit   2 bit    3     4 bit  blinking, multiplexing
+ *   Bt453   66MHz  8 bit   2 bit    3     8 bit
+ *   Bt454  170MHz  4 bit   1 bit    3     4 bit
+ *   Bt455  170MHz  4 bit   1 bit    1     4 bit
+ *   Bt457  165Mhz  8 bit   2 bit    1     8 bit  blinking, multiplexing
+ *   Bt458  165MHz  8 bit   2 bit    3     8 bit  blinking, multiplexing
+ *
+ * Reference: http://www.bitsavers.org/components/brooktree/_dataBooks/1991_Brooktree_Product_Databook.pdf
+ *
+ * The bt45x_mono_device_base uses the standard red/green/blue read/write
+ * cycles defined in the databook, with color data active on the green cycle.
+ *
+ * TODO
+ *   - refactor to separate devices with registers
+ *   - implement blinking and overlay
+ *   - unsure about address masking when accessing overlay colors
+ */
+
+#include "emu.h"
+#include "bt45x.h"
+
+#define LOG_GENERAL (1U << 0)
+#define LOG_READS   (1U << 1)
+
+#define VERBOSE (0)
+
+#include "logmacro.h"
+
+DEFINE_DEVICE_TYPE(BT451, bt451_device, "bt451", "Brooktree 125/135/165 MHz Monolithic CMOS 256 Color Palette RAMDAC")
+DEFINE_DEVICE_TYPE(BT453, bt453_device, "bt453", "Brooktree 66 MHz Monolithic CMOS 256 Color Palette RAMDAC")
+DEFINE_DEVICE_TYPE(BT454, bt454_device, "bt454", "Brooktree 170 MHz Monolithic CMOS 16 Color Palette RAMDAC")
+DEFINE_DEVICE_TYPE(BT455, bt455_device, "bt455", "Brooktree 170 MHz Monolithic CMOS 16 Color Palette RAMDAC")
+DEFINE_DEVICE_TYPE(BT457, bt457_device, "bt457", "Brooktree 125/135/165 MHz Monolithic CMOS 256 Color Palette RAMDAC")
+DEFINE_DEVICE_TYPE(BT458, bt458_device, "bt458", "Brooktree 125/135/165 MHz Monolithic CMOS 256 Color Palette RAMDAC")
```

Comment:
The full names don't include the BT45x part number.  I kinda think they should.  Also does it really need "Monolithic CMOS" and "Palette" in the full names?  They're pretty long.

Reply:
Yeah, "Brooktree BT45x y Color RAMDAC" would be quite sufficient.

Reply:
For lack of any other specific guidelines, I've been using the name/title printed on the datasheets. Of course abbreviating these is possible, but a bit arbitrary. No problem to do it though - I'll also do so with the other one.